user-guide/Troubleshooting/Procedures/Cassandra_expired_blocking_SSTables.md

```diff
@@ -0,0 +1,49 @@
+---
+uid: Cassandra_expired_blocking_SSTables
+---
+
+# Expired SSTables blocking other tables from being dropped
+
+In case the only data SSTables in Cassandra contain is tombstones, and the tables are therefore expired, they could block other tables that are also expired from being dropped.
```

Comment:
Are there any symptoms of this issue that users would recognize, so that they know that this is the issue they are facing?

Reply:
Yes there are. I realized by first seeing an increase in the disk usage of the data partition of Cassandra. The TWCS keyspaces (trend_data_medium, trend_data_short and trend_data_long) where showing a memory leak behavior, and they never seemed to drop.

I used the cassandra-tools to double check what was going on. The commands I used for installing and finding out what was the issue were the following

```
yum install cassandra-tools	 

sudo systemctl stop cassandra  

sstableexpiredblockers zm1_cc_trend_data_short trend_data_short   

for f in *Data.db; do meta=$(sudo sstablemetadata $f); echo -e "Max:" $(date --date=@$(echo "$meta" | grep Maximum\ time | cut -d" " -f3| cut -c 1-10) '+%m/%d/%Y') "Min:" $(date --date=@$(echo "$meta" | grep Minimum\ time | cut -d" " -f3| cut -c 1-10) '+%m/%d/%Y') $(echo "$meta" | grep droppable) ' \t ' $(ls -lh $f | awk '{print $5" "$6" "$7" "$8" "$9}'); done | sort 
``` 

With the output of the sstableexpiredblockers and the for below it can be seen which tables are blocking others from being dropped. Basically, if the percetange of tombstones in the SSTable is higher than 1, it can be assumed that all data is tombstone data.